dexcom/event.go

```diff
@@ -196,44 +217,69 @@ func (e *Event) Validate(validator structure.Validator) {
 }
 
 func (e *Event) validateCarbs(validator structure.Validator) {
 	validator.String("eventSubType", e.SubType).NotExists()
-	if e.Unit != nil || e.Value != nil {
-		validator.String("unit", e.Unit).Exists().OneOf(EventUnitCarbsGrams)
-		validator.Float64("value", e.Value).Exists().InRange(EventValueCarbsGramsMinimum, EventValueCarbsGramsMaximum)
+	validator.String("unit", e.Unit).Exists().OneOf(EventUnitCarbsGrams)
+	validator.String("value", e.Value).Exists().NotEmpty()
+	if e.Value != nil {
+		floatVal, _ := strconv.ParseFloat(*e.Value, 64)
+		validator.Float64("value", &floatVal).Exists().InRange(EventValueCarbsGramsMinimum, EventValueCarbsGramsMaximum)
 	}
 }
 
 func (e *Event) validateExercise(validator structure.Validator) {
-	// HACK: Dexcom - value of -1 is invalid; ignore unit and value instead (per Dexcom)
-	if e.Value != nil && *e.Value == -1.0 {
-		e.Unit = nil
-		e.Value = nil
-	}
-
 	validator.String("eventSubType", e.SubType).OneOf(EventSubTypesExercise()...)
-	if e.Unit != nil || e.Value != nil {
-		validator.String("unit", e.Unit).Exists().OneOf(EventUnitExerciseMinutes)
-		validator.Float64("value", e.Value).Exists().InRange(EventValueExerciseMinutesMinimum, EventValueExerciseMinutesMaximum)
+	validator.String("unit", e.Unit).Exists().OneOf(EventUnitExerciseMinutes)
+	validator.String("value", e.Value).Exists().NotEmpty()
+	if e.Value != nil {
+		floatVal, _ := strconv.ParseFloat(*e.Value, 64)
+		validator.Float64("value", &floatVal).Exists().InRange(EventValueExerciseMinutesMinimum, EventValueExerciseMinutesMaximum)
 	}
 }
 
 func (e *Event) validateHealth(validator structure.Validator) {
 	validator.String("eventSubType", e.SubType).OneOf(EventSubTypesHealth()...)
-	validator.String("unit", e.Unit).NotExists()
-	validator.Float64("value", e.Value).EqualTo(0)
+	validator.String("value", e.Value).Exists().NotEmpty()
+}
+
+func (e *Event) validateNote(validator structure.Validator) {
+	validator.String("value", e.Value).Exists().NotEmpty()
+}
+
+func (e *Event) validateUnkown(validator structure.Validator) {
+	validator.String("value", e.Value).Exists().NotEmpty()
 }
 
 func (e *Event) validateInsulin(validator structure.Validator) {
 	validator.String("eventSubType", e.SubType).OneOf(EventSubTypesInsulin()...)
-	if e.Unit != nil || e.Value != nil {
-		validator.String("unit", e.Unit).Exists().OneOf(EventUnitInsulinUnits)
-		validator.Float64("value", e.Value).Exists().InRange(EventValueInsulinUnitsMinimum, EventValueInsulinUnitsMaximum)
+	validator.String("unit", e.Unit).Exists().OneOf(EventUnitInsulinUnits)
+	validator.String("value", e.Value).Exists().NotEmpty()
+	if e.Value != nil {
+		floatVal, _ := strconv.ParseFloat(*e.Value, 64)
+		validator.Float64("value", &floatVal).Exists().InRange(EventValueInsulinUnitsMinimum, EventValueInsulinUnitsMaximum)
+	}
+}
+
+func (e *Event) validateBG(validator structure.Validator) {
+	validator.String("unit", e.Unit).Exists().OneOf(EventUnitMgdL)
```

Comment:
Just a note to comment that I find it a bit odd they only appear to allow `mg/dL` for this type.

Reply:
(in the API docs, that is)

Reply:
yeah ... I am wondering if this is caught up in a future proofing / migration for units. [In the docs ](https://developer.dexcom.com/docs/dexcomv3/endpoint-overview/#units) it says that units are specified with the response BUT also that 

>  At the moment, all glucose values are reported in units of mg/dL and trend rates in units of mg/dL/min